.github/workflows/codeql.yml

```diff
@@ -0,0 +1,126 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", "master" ]
+  schedule:
+    - cron: '0 0 * * *'
+  pull_request:
+    branches: '*'
+
+jobs:
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: ubuntu-22.04
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'c-cpp' ]
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
```

Comment:
This link has a list of supported compilers which (to no surprise) doesn't include SDCC. Does running CodeQL on this codebase even work?

Reply:
It works, but only scans 11% of C source files.
![image](https://github.com/system76/ec/assets/20792991/3ca6e242-8ee3-4aa7-8c2a-cc355b2e2fe8)

Reply:
Found some minor issues.
![image](https://github.com/system76/ec/assets/20792991/98d2ba88-1a0c-4145-9c67-f13c8db1878c)

Reply:
Right, so it only works with avr-gcc that we use for the Mega2560 dev tool.